pkg/controller/controller.go

```diff
@@ -522,3 +522,27 @@ func isInstanceReady(instance *v1alpha1.Instance) bool {
 
 	return false
 }
+
+func isOwnedByBinding(objectMeta *metav1.ObjectMeta, binding *v1alpha1.Binding) bool {
+	ref := getControllerRef(objectMeta)
+	if ref == nil {
+		return false
+	}
+	return ref.APIVersion == binding.APIVersion &&
+		ref.Kind == binding.Kind &&
+		ref.Name == binding.Name &&
```

Comment:
@nilebox you need to set all the fields, but you only really have to check UID actually because it is unique. Name is also checked because it makes sense. On the other hand we shouldn't check APIVersion (i.e. group and version) and Kind because versions are just a view for the same data i.e. it is the same object actually. Kind and group is a bit tricky - as it turned out there is a notion of co-habitating types! Deployment is one of them. i.e. Deployment in extensions/v1beta1 and in apps/v1beta1 are backed by the same object internally! I was not expecting that but it's how it works. I guess Kind may be changed too in such "migration" cases. Also I've checked Deployment controller and some other code as an example an it does not check those things.